src/controllers/expenseController.js

```diff
@@ -0,0 +1,94 @@
+const expensesService = require('../services/expensesService');
+const userService = require('../services/userService');
+
+const get = async (req, res) => {
+  const { userId, categories, from, to } = req.query;
+  const filteredExpenses = await expensesService.getAllExpenses(
+    userId,
+    categories,
+    from,
+    to,
+  );
+
+  res.status(200).send(filteredExpenses);
+};
+
+const getOne = async (req, res) => {
+  const { id } = req.params;
+  const expense = await expensesService.getExpenseById(id);
+
+  if (!expense) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  res.send(expense);
+};
+
+const create = async (req, res) => {
+  const { userId, spentAt, title, amount, category, note } = req.body;
+  const user = await userService.getUserById(userId);
+
+  if (!user) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const expense = await expensesService.createExpense(
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+  );
+
+  res.statusCode = 201;
+  res.send(expense);
+};
+
+const remove = async (req, res) => {
+  const { id } = req.params;
+  const expense = await expensesService.getExpenseById(id);
+
+  if (!expense) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  await expensesService.deleteExpense(id);
+
+  res.sendStatus(204);
+};
+
+const update = async (req, res) => {
+  const { id } = req.params;
+  const expense = await expensesService.getExpenseById(id);
+
+  if (!expense) {
+    res.sendStatus(404);
```

Comment:
In the `update` function, you are sending a 404 status code if the expense is not found. This is correct, but you should also consider handling the case where the update operation fails for reasons other than the expense not being found. For example, if the update operation throws an error, you could catch that error and send a 500 Internal Server Error status code.